modules/toxiproxy/src/main/java/org/testcontainers/containers/ToxiproxyContainer.java

```diff
@@ -0,0 +1,115 @@
+package org.testcontainers.containers;
+
+import com.github.dockerjava.api.command.InspectContainerResponse;
+import eu.rekawek.toxiproxy.Proxy;
+import eu.rekawek.toxiproxy.ToxiproxyClient;
+import eu.rekawek.toxiproxy.model.ToxicDirection;
+import eu.rekawek.toxiproxy.model.ToxicList;
+import org.testcontainers.containers.wait.strategy.HttpWaitStrategy;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ * Container for resiliency testing using <a href="https://github.com/Shopify/toxiproxy">Toxiproxy</a>.
+ */
+public class ToxiproxyContainer extends GenericContainer<ToxiproxyContainer> {
+
+    private static final String IMAGE_NAME = "shopify/toxiproxy:2.1.0";
+    private static final int TOXIPROXY_CONTROL_PORT = 8474;
+    private static final int FIRST_PROXIED_PORT = 8666;
+    private static final int LAST_PROXIED_PORT = 8666 + 31;
+
+    private ToxiproxyClient client;
+    private final Map<String, ContainerProxy> proxies = new HashMap<>();
+    private final AtomicInteger nextPort = new AtomicInteger(FIRST_PROXIED_PORT);
+
+    public ToxiproxyContainer() {
+        super(IMAGE_NAME);
+    }
+
+    public ToxiproxyContainer(String imageName) {
+        super(imageName);
+    }
+
+    @Override
+    protected void configure() {
+        addExposedPorts(TOXIPROXY_CONTROL_PORT);
```

Comment:
I would change `ToxiproxyContainer()` to call `this(IMAGE_NAME)` and put the customizations in a constructor.
Why?

Because if somebody will want to override some of them, it won't be possible to do with the fluent API unless you create your own class and override `configure`

Reply:
Yes - I should have remembered this exact problem with the Postgres container class.